ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -145,6 +118,44 @@ static std::vector<std::string> split_string(const std::string& str,
   return strings;
 }
 
+/// @brief Split an action definition into individual goal, result and feedback definitions.
+/// @param action_definition The full action definition as read from a .action file
+/// @return A tuple holding goal, result and feedback definitions
+static std::tuple<std::string, std::string, std::string> split_action_msg_definition(
+  const std::string& action_definition) {
+  constexpr char SEP[] = "---";
+
+  const auto definitions = split_string(action_definition, SEP);
```

Comment:
What if one of the segments has a string constant with `---` in it? e.g. `string FOO=---` ?